internal/crypto/helpers.go

```diff
@@ -116,11 +116,30 @@ func ValidateRequestCerts(r *http.Request, certs []*tls.Certificate) error {
 			// Happy flow, we matched a certificate
 			return nil
 		}
+
+		if cert.Leaf.IsCA && verifyCertAgainstCA(cert, peerCertificate) {
+			return nil
+		}
 	}
 
 	return errors.New("Certificate with SHA256 " + certID + " not allowed")
 }
 
+func verifyCertAgainstCA(caCert *tls.Certificate, peerCert *x509.Certificate) bool {
+	// Create a certificate pool and add the CA certificate to it
+	roots := x509.NewCertPool()
```

Comment:
Would we want to keep opts/roots around in memory so they can be reused between requests?

Reply:
Upon thinking about it again, the client certificates attached in the API would be used during TLS handshake and this additional check against certificate authority isn't required when the uploaded certificate is a CA certificate. 
Also, we could do an improvement by fetching the client certificates during API reload and caching it in middleware struct itself